api/insight/apimiddleware.go

```diff
@@ -91,3 +109,161 @@ func (c *insightApiContext) PostBroadcastTxCtx(next http.Handler) http.Handler {
+func (c *insightApiContext) GetAddressCtx(r *http.Request) string {
+	address, ok := r.Context().Value(ctxAddress).(string)
+	if !ok {
+		apiLog.Trace("address not set")
+		return ""
+	}
+	return address
+}
+
+// GetToCtx retrieves the ctxTo data ("to") from the request context. If not
+// set, the return value ok is false.
+func (c *insightApiContext) GetToCtx(r *http.Request) (int64, bool) {
+	to, ok := r.Context().Value(ctxTo).(int)
+	if !ok {
+		apiLog.Errorf("to param not set")
+		return int64(0), false
+	}
+	return int64(to), true
+}
+
+// GetFromCtx retrieves the ctxFrom data ("from") from the request context.
+// If not set, the return value ok is false.
+func (c *insightApiContext) GetFromCtx(r *http.Request) (int64, bool) {
+	from, ok := r.Context().Value(ctxFrom).(int)
+	if !ok {
+		apiLog.Errorf("from param not set")
+		return int64(0), false
+	}
+	return int64(from), true
+}
+
+// GetNoAsmCtx retrieves the ctxNoAsm data ("noAsm") from the request context.
+// If not set, the return value is false.
+func (c *insightApiContext) GetNoAsmCtx(r *http.Request) bool {
+	noAsm, ok := r.Context().Value(ctxNoAsm).(bool)
+	if !ok {
+		return false
+	}
+	return noAsm
+}
+
+// GetNoScriptSigCtx retrieves the ctxNoScriptSig data ("noScriptSig") from the
+// request context. If not set, the return value is false.
+func (c *insightApiContext) GetNoScriptSigCtx(r *http.Request) bool {
+	noScriptSig, ok := r.Context().Value(ctxNoScriptSig).(bool)
+	if !ok {
+		return false
+	}
+	return noScriptSig
+}
+
+// GetNoSpentCtx retrieves the ctxNoSpent data ("noSpent") from the
+// request context. If not set, the return value is false.
+func (c *insightApiContext) GetNoSpentCtx(r *http.Request) bool {
+	noNoSpent, ok := r.Context().Value(ctxNoSpent).(bool)
+	if !ok {
+		return false
+	}
+	return noNoSpent
+}
+
+// PaginationCtx will parse the query parameters for from/to values.
+func (c *insightApiContext) PaginationCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		to, from := r.FormValue("to"), r.FormValue("from")
+		fromint, err := strconv.Atoi(from)
+		if err == nil {
+			ctx = context.WithValue(r.Context(), ctxFrom, fromint)
+		}
+		toint, err := strconv.Atoi(to)
+		if err == nil {
+			ctx = context.WithValue(ctx, ctxTo, toint)
+		}
+		next.ServeHTTP(w, r.WithContext(ctx))
+	})
+}
+
+// Process params given in post body for an addrs endpoint
+func (c *insightApiContext) PostAddrsTxsCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var req apitypes.InsightMultiAddrsTx
+		var from, to, noAsm, noScriptSig, noSpent int64
+		body, err := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		if err != nil {
+			apiLog.Errorf("error reading JSON message: %v", err)
+			writeInsightError(w, fmt.Sprintf("error reading JSON message: %v", err))
+			return
+		}
+		err = json.Unmarshal(body, &req)
```

Comment:
Do you have an example of the type of request (header and body) that we should expect for this?

Reply:
should be POST and "application/json"

```go
package main

import (
	"fmt"
	"strings"
	"net/http"
	"io/ioutil"
)

func main() {

	url := "http://127.0.0.1:7777/insight/api/addrs/utxo"

	payload := strings.NewReader("{\"addrs\":\"Dca7Vsv42RAJC6cEdw7dyhLER8QJCHiUYNL,DsTHdW8erZsD1Fqt8uCfapVR1Qsf9KaMwBN\"}")

	req, _ := http.NewRequest("POST", url, payload)

	req.Header.Add("Content-Type", "application/json")
	req.Header.Add("Cache-Control", "no-cache")

	res, _ := http.DefaultClient.Do(req)
}
```

Reply:
Awesome thanks.  How about content-length or content-encoding.  I imagine we want to ensure content-encoding is not chunked, and with content-length set, set to something we threshold

Reply:
Ok, I checked the content-length.    Transfer encoding?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding